### PR TITLE
Adds anomaly detection, lint fixes, and bumps to 0.3.0

### PR DESCRIPTION
**Pull request by ashenfad**
This PR adds anomaly detection and does a little house cleaning.

### CODE REVIEW

File: src/bigml/api/anomaly_score.clj [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+(ns bigml.api.anomaly-score
+  "Functions for scoring with BigML anomaly detectiors.
+     https://bigml.com/api/anomalyscores"
+  (:require (bigml.api [core :as api]
+                       [utils :as utils]))
+  (:refer-clojure :exclude [list]))
+
+(defn create
+  "Creates an anomaly detector from an anomaly detector. A detector
```
Comment (jaor) [inline src/bigml/api/anomaly_score.clj]: you meant creates an anomaly score?

File: src/bigml/api/anomaly_score.clj [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        (op-fn (inputs index))))))
+
+(defn- predicates-fn [predicates id->index]
+  (let [pred-fns (mapv #(predicate-fn % id->index) predicates)
+        pred-count (count predicates)]
+    (fn [inputs]
+      (loop [i 0]
+        (if (and (< i pred-count) ((pred-fns i) inputs))
+          (recur (inc i))
+          i)))))
```
Comment (jaor) [inline src/bigml/api/anomaly_score.clj]: i would have written `(count (take-while #(% inputs) pred-fns))`... are you worried about performance?

File: src/bigml/api/anomaly_score.clj [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+              (if normalize-repeats
+                (+ depth repeat-depth)
+                depth)
+              (< 0 i pred-count) (dec depth)
+              (and (= i pred-count) (seq children))
+              (loop [i 0]
+                (if (< i children-count)
+                  (or ((children i) inputs normalize-repeats)
+                      (recur (inc i)))
+                  depth)))))))
```
Comment (jaor) [inline src/bigml/api/anomaly_score.clj]: this loop also seems doable with a single take-while...

File: src/bigml/api/anomaly_score.clj (hunk 1 above)
Comment (ashenfad) [inline src/bigml/api/anomaly_score.clj]: I did - fixed! d9b2f9d

File: src/bigml/api/anomaly_score.clj (hunk 2 above)
Comment (ashenfad) [inline src/bigml/api/anomaly_score.clj]: I originally did just that. But liked you guessed, it's slower than `loop`. The loops here and in `node-fn` are used so frequently that the time savings were worth it.

**Comment by ashenfad** [comment]
Doc fix is in and I'm leaving the loops for speed.

**Comment by ashenfad** [comment]
Hold on a moment, I'm making a readme fix for referencing the clojars version.

**Comment by ashenfad** [comment]
Okay, it seems the expected format for clojars badges had changed. Fixed now.

**Comment by jaor** [comment]
looks good. merged and tagged!

**Pull request merged**